src/java.base/share/classes/java/lang/reflect/ProxyGenerator.java

```diff
@@ -882,7 +819,7 @@ private void codeClassForName(CodeBuilder cob, Class<?> cl) {
             cob.ldc(cl.getName())
                .iconst_0() // false
                .aload(0)// classLoader
-               .invokestatic(MRE_Class_forName);
+               .invokestatic(CD_Class, "forName", MTD_Class_String_boolean_ClassLoader);
```

Comment:
We can probably replace this `forName(name, false, thisClassLoader)` with loading a class constant to reduce load on symbols.

Reply:
I'm wondering why all the `Class.forName(... <proxyClass>.getClassLoader())` is necessary?
Simple `ldc <class entry>` seems to work well.

Reply:
Indeed, I initially added it because a JBS ticket recommended so. ldc instruction has the same behavior as such as forName call.